astropy/io/fits/hdu/groups.py

```diff
@@ -503,7 +503,7 @@ def _verify(self, option='warn'):
 
         self.req_cards('GCOUNT', pos, _is_int, 1, option, errs)
         self.req_cards('PCOUNT', pos, _is_int, 0, option, errs)
-        self.req_cards('GROUPS', pos, lambda v: (v == True), True, option,
+        self.req_cards('GROUPS', pos, lambda v: (v), True, option,
```

Comment:
My original changes made no sense, this should always return a bool, thus changed it to be ``is True``. And I'm pretty sure that the input is never an array (as being the value of the given header keyword).